hotshot-state-prover/src/bin/state-prover.rs

```diff
@@ -0,0 +1,132 @@
+    #[clap(long, env = "ESPRESSO_SEQUENCER_ETH_MNEMONIC", default_value = None)]
+    eth_mnemonic: String,
+
+    /// Index of a funded account derived from eth-mnemonic.
+    #[clap(
+        long,
+        env = "ESPRESSO_SEQUENCER_ETH_ACCOUNT_INDEX",
+        default_value = "0"
+    )]
+    eth_account_index: u32,
+
+    /// Number of nodes in the stake table.
+    /// WARNING: This is used temporarily to initialize a static stake table.
+    ///          In the future we should get the stake table from the contract.
+    #[clap(
+        short,
+        long,
+        env = "ESPRESSO_ORCHESTRATOR_NUM_NODES",
+        default_value = "5"
+    )]
+    num_nodes: usize,
+
+    /// Seed to use for generating node keys.
+    /// WARNING: This is used temporarily to initialize a static stake table.
+    ///          In the future we should get the stake table from the contract.
+    #[arg(long, env = "ESPRESSO_ORCHESTRATOR_KEYGEN_SEED", default_value = "0x0000000000000000000000000000000000000000000000000000000000000000", value_parser = parse_seed)]
+    keygen_seed: [u8; 32],
+}
+
+#[derive(Clone, Debug, Snafu)]
+pub struct ParseDurationError {
+    reason: String,
+}
+
+fn parse_duration(s: &str) -> Result<Duration, ParseDurationError> {
+    ClDuration::from_str(s)
+        .map(Duration::from)
+        .map_err(|err| ParseDurationError {
+            reason: err.to_string(),
+        })
+}
+
+#[derive(Debug, Snafu, From)]
+enum ParseSeedError {
+    #[snafu(display("seed must be valid hex: {source}"))]
+    Hex { source: FromHexError },
+
+    #[snafu(display("wrong length for seed {length} (expected 32)"))]
+    WrongLength { length: usize },
+}
+
+fn parse_seed(s: &str) -> Result<[u8; 32], ParseSeedError> {
+    <[u8; 32]>::try_from(hex::decode(s)?)
+        .map_err(|vec| ParseSeedError::WrongLength { length: vec.len() })
+}
+
+#[async_std::main]
+async fn main() {
+    setup_logging();
+    setup_backtrace();
+
+    let args = Args::parse();
+
+    // prepare config for state prover from user options
+    let provider = Provider::<Http>::try_from(args.l1_provider.to_string()).unwrap();
+    let chain_id = provider.get_chainid().await.unwrap().as_u64();
+    let config = StateProverConfig {
+        relay_server: args.relay_server.clone(),
+        update_interval: args.update_interval,
+        l1_provider: args.l1_provider.clone(),
+        light_client_address: args.light_client_address,
+        eth_signing_key: MnemonicBuilder::<English>::default()
+            .phrase(args.eth_mnemonic.as_str())
+            .index(args.eth_account_index)
+            .expect("error building wallet")
+            .build()
+            .expect("error opening wallet")
+            .with_chain_id(chain_id)
+            .signer()
+            .clone(),
+        num_nodes: args.num_nodes,
+        seed: args.keygen_seed,
+    };
+
+    if args.daemon {
+        // Launching the prover service daemon
+        run_prover_service(config).await;
```

Comment:
I haven't reviewed everything in detail so might have overlooked it. Does the prover currently have a health check API? It's used to confirm that the service is running and restart it if necessary. If not, adding a tide-disco server will provide a simple health check.

https://github.com/EspressoSystems/espresso-sequencer/blob/58852cbd989606f103a41664ef4c728859de007d/sequencer/src/bin/commitment-task.rs#L138

Reply:
health check added [8601b08](https://github.com/EspressoSystems/espresso-sequencer/pull/1041/commits/8601b080bd0c086bbfd0e7f2920e289940b5f36f)